client/ayon_blender/plugins/publish/validate_absolute_external_data_paths.py

```diff
@@ -0,0 +1,61 @@
+import bpy
+import pyblish.api
+from ayon_core.pipeline import (
+    OptionalPyblishPluginMixin
+)
+from ayon_core.pipeline.publish import (
+    RepairAction,
+    ValidateContentsOrder,
+    PublishValidationError
+)
+
+
+
+class ValidateAbsoluteExternalDataPaths(pyblish.api.InstancePlugin,
+                                        OptionalPyblishPluginMixin):
+    """Validates Absolute External Data Paths
+
+    This validator checks if all external data paths are absolute
+    to ensure the links would not be broke when publishing
+    """
+
+    label = "Validate Absolute External Data Paths"
+    order = ValidateContentsOrder
+    hosts = ["blender"]
+    optional = True
+    actions = [RepairAction]
+
+    @classmethod
+    def get_invalid(cls, instance):
+        invalid = []
+        object_type = type(bpy.data.objects)
+        for attr in dir(bpy.data):
+            collections = getattr(bpy.data, attr)
+            if isinstance(collections, object_type):
+                for data_block in collections:
+                    if hasattr(data_block, "filepath") and data_block.filepath:
+                        if data_block.filepath != bpy.path.abspath(data_block.filepath):
+                            cls.log.error(f"Absolute external data path {data_block.filepath} "
+                                          "is not absolute path")
```

Comment:
```suggestion
                            cls.log.error(f"External data path {data_block.filepath} "
                                          "is not absolute path")
```

Also, I wonder if referring to these as "Data block filepath" makes more sense since that's apparently what blender calls them instead of "External data path"?